### PR TITLE
validate_phone_datetime supports datetime.date

### PR DESCRIPTION
**Pull request by kaapstorm**
Forms (submitted in 2015 and 2016, and probably others) have case blocks whose "date_modified" attributes are `datetime.date` instances. Migrating these forms is raising `PhoneDateValueError`.

This change allows `validate_phone_datetime()` to support `datetime.date` values the same way it supports date string values.

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class ValidatePhoneDatetimeTests(SimpleTestCase):
+
+    def test_datetime_string(self):
+        datetime_string = '2019-08-27T17:50:00.000'
+        result = validate_phone_datetime(datetime_string)
+
+        datetime_ = iso8601.parse_date(datetime_string)
+        assert datetime_ == datetime.datetime(
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/tests/test_util.py]: This is confusing - why do you need to assert that this is true? Also you could just use the manually created date in the `assertEqual` below if you don't trust the value of `datetime_`

Comment (kaapstorm) [reply corehq/ex-submodules/casexml/apps/case/tests/test_util.py]: My thought here was about using "tests as documentation", to show the reader what is happening. `validate_phone_datetime()` uses `iso8601.parse_date()`. So this `assert` was supposed to be like "Look, this is what we get." ...

But I agree, especially re-reading this, this is confusing. There is no explanation for what that's doing there, and why, and why I used `assert` here and `self.assertEqual()` on the next line. I'll drop this.

**Comment by dannyroberts** [comment]
Does this also make it so that you can submit a new form with dates instead of datetimes and it will be accepted? Is that something we want? Honestly not sure, but thought it's worth addressing explicitly

**Comment by kaapstorm** [comment]
That's a very good question. Currently `validate_phone_datetime()` supports dates as long as they are passed as strings and not `datetime.date` instances. I am not sure when we pass parameters as strings instead of as objects, but if we get strings from new form submissions, then we might be doing this already.

... Although, I'm not sure that's a good thing.

**Comment by millerdev** [comment]
Could we fairly easily address most of @dannyroberts concerns by making `validate_phone_datetime()` only convert date to datetime if the date is far in the past, say before 2017?

The only downside I see to this solution is that a newly submitted form could "forge" some values that are expected to be datetimes by using dates that are far in the past (in this example, pre-2017). I'm not sure this is a big concern. If we wanted to make `validate_phone_datetime()` immune to that kind of attack then we'd need to give it more context, like have it be aware of the submission `received_on` value, but that seems way too hard and probably not worth it.

**Comment by kaapstorm** [comment]
We only needed this in the context of migrating one specific domain. I'm going to close this PR for now because YAGNI, but I won't delete the branch. If it ever comes up again in another context, the code will be there and we can revisit this.

**Pull request closed without merging**